blog/_posts/2023-01-24-scala-center-2023-roadmap.md

```diff
@@ -0,0 +1,245 @@
+grateful to all of them.
+
+In the following subsections we remind our ongoing and recurring projects, and we present our most important goals as
+well as some additional stretch goals that would need more resources.
+
+The roadmap we present here is of course subject to adjustments throughout the year.
+
+### Language, Compiler, Standard Library
+
+In addition to keeping the Scala Improvement Process going, and reducing the number of bugs in the compiler and standard
+library, we will:
+
+- **Make the compiler error messages clearer and more actionable.** The compiler should help developers write correct code
+  instead of “just complaining” about incorrect code. You can find a list of related issues
+  [here](https://github.com/lampepfl/dotty/issues?q=is%3Aopen+is%3Aissue+label%3Aarea%3Areporting).
+- **Bring the compiler’s suggestions into the IDEs.** Instead of emitting only text output, the compiler
+  will produce structured output that IDEs (Metals and IntelliJ) will be able to read to automatically provide “quick
+  fixes” to the users. This should reduce the implementation (and maintenance) costs on the IDE side.
+- **Unfreeze the Scala standard library.** The standard library has not changed (except for bug fixes or some performance
+  improvements) since Scala 2.13, which was released almost four years ago. [Dropping the forwards binary compatibility
+  requirement](https://github.com/scala/improvement-proposals/pull/54) would be a first step to allow the introduction
+  of new classes or the addition of new members to the existing classes. Then, we will create a process to validate
+  what should be in the standard library and what should stay outside of it.
+- **Clarify and communicate the Scala 3 roadmap.** We will coordinate with the compiler team to clarify what will
+  go into the [Long-Term Support version]({% link blog/_posts/2022-08-17-long-term-compatibility-plans.md %})
+  of Scala 3, and make sure that roadmap is well communicated. We will focus primarily on the items that will
+  enable more users to migrate from Scala 2 to Scala 3 (e.g., missing compiler options).
+
+Our stretch goals include: publish a formal specification for Scala 3, investigate opportunities for reducing
+compilation times, implement multi-threading in Scala Native, and remove the dependency to the Google Closure
+Compiler in Scala.js.
+
+### Documentation and MOOCs
+
+In addition to maintaining the Scala website and managing our online course learners, we will:
+
+- **Add “task-oriented” content to the documentation.** Currently, the documentation is rather “language features oriented”,
+  but it does not really help solve concrete problems. We will select a set of libraries (the “Scala Toolkit”) and we
+  will write tutorials showing how to perform common programming tasks (working with JSON, writing tests, etc.).
+  We will use libraries created by the community and that do not require advanced knowledge of Scala.
```

Comment:
It may be good also to mention that we will actively coordinate and help with the maintenance of the libraries included in the toolkit.

Reply:
OK, done in 4865541e